assets/dev/js/editor/controls/base-data.js

```diff
@@ -229,9 +229,14 @@ ControlBaseDataView = ControlBaseView.extend( {
 			cleanValue = this.getCleanControlValue();
 
 		if ( cleanValue instanceof Object ) {
-			return Object.keys( cleanValue ).length ?
-				Object.assign( {}, parentValue, cleanValue ) :
-				parentValue;
+			const values = Object.assign(
+				{},
+				this.model.get( 'placeholder' ),
+				parentValue,
+				cleanValue
+			);
+
+			return Object.keys( values ).length ? values : undefined;
 		}
 
 		return this.getControlValue() || parentValue;
```

Comment:
If I get the idea, there is a hierarchy for the placeholders, first, it checks for `clean value`, then for `parent value` and at the end for `placeholder` am I right? is this behavior applied only for object values or also for primitive values?

can you write a unit test specific for this method to cover all the cases?